src/Concerns/HasTimezonedTimestamps.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Whitecube\LaravelTimezones\Concerns;
+
+use Whitecube\LaravelTimezones\Casts\TimezonedDatetime;
+
+trait HasTimezonedTimestamps
+{
+    /**
+     * Determine if the given attribute is a date or date castable.
+     *
+     * @param  string  $key
+     * @return bool
+     */
+    protected function isDateAttribute($key)
+    {
+        return (in_array($key, $this->getDates(), true) ||
+            $this->isDateCastable($key)) &&
+            ! $this->hasTimezonedDatetimeCast($key);
+    }
+
+    /**
+     * Check if key is a timezoned datetime cast
+     * 
+     * @param string $key 
+     * @return bool 
+     */
+    protected function hasTimezonedDatetimeCast(string $key): bool
+    {
+        $casts = $this->getCasts();
+
+        return array_key_exists($key, $casts) && $casts[$key] === TimezonedDatetime::class;
```

Comment:
```suggestion
        return in_array($casts[$key] ?? null, [TimezonedDatetime::class, ImmutableTimezonedDatetime::class]);
```

Reply:
Wait... this isn't enough either since one can cast using a custom format, which would break this checkup : 

```php
protected $casts = [
        'created_at' => ImmutableTimezonedDatetime::class . 'H:i m/d/Y',
    ]
```

Reply:
This is probably better (didn't test) : 

```php
$cast = $this->getCasts()[$key] ?? null;

if(! $cast) return false;

return in_array(explode(':', $cast)[0], [TimezonedDatetime::class, ImmutableTimezonedDatetime::class]);
```